adapters/globalsun/globalsun.go

```diff
@@ -0,0 +1,148 @@
+package globalsun
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type reqBodyExt struct {
+	GlobalsunBidderExt reqBodyExtBidder `json:"bidder"`
+}
+
+type reqBodyExtBidder struct {
+	Type        string `json:"type"`
+	PlacementID string `json:"placementId,omitempty"`
+	EndpointID  string `json:"endpointId,omitempty"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var err error
+	adapterRequests := make([]*adapters.RequestData, 0, len(request.Imp))
+
+	for _, imp := range request.Imp {
+		reqCopy := *request
```

Comment:
Nitpick: aside from its `reqCopy.Imp` field, `reqCopy` doesn't seem to get modified in any other way. I belive this opens the oportunity to slightly optimize if we create this shallow copy once:
```
36   func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
37       var err error
38       adapterRequests := make([]*adapters.RequestData, 0, len(request.Imp))
39   
   +     reqCopy := *request
40       for _, imp := range request.Imp {
41 -         reqCopy := *request
42           reqCopy.Imp = []openrtb2.Imp{imp}
43   
```